tests/codemods/base_codemod_test.py

```diff
@@ -19,43 +16,84 @@ class BaseCodemodTest:
-        wrapper = cst.MetadataWrapper(input_tree)
-        codemod_instance = self.codemod(
-            CodemodContext(wrapper=wrapper),
-            self.file_context,
-        )
-        return codemod_instance
+        self.file_context = None
 
-    def run_and_assert(self, tmpdir, input_code, expected):
-        tmp_file_path = Path(tmpdir / "code.py")
-        self.run_and_assert_filepath(tmpdir, tmp_file_path, input_code, expected)
+    def run_and_assert(  # pylint: disable=too-many-arguments
+        self,
+        tmpdir,
+        input_code,
+        expected,
+        num_changes: int = 1,
+        root: Path | None = None,
+        files: list[Path] | None = None,
+    ):
+        root = root or tmpdir
+        tmp_file_path = files[0] if files else Path(tmpdir) / "code.py"
+        tmp_file_path.write_text(dedent(input_code))
+
+        files_to_check = files or [tmp_file_path]
 
-    def run_and_assert_filepath(self, root, file_path, input_code, expected):
-        input_tree = cst.parse_module(dedent(input_code))
         self.execution_context = CodemodExecutionContext(
             directory=root,
-            dry_run=True,
+            dry_run=False,
             verbose=False,
             registry=mock.MagicMock(),
             repo_manager=mock.MagicMock(),
+            path_include=[f.name for f in files_to_check],
+            path_exclude=[],
         )
-        self.file_context = FileContext(
-            root,
-            file_path,
-            [],
-            [],
-            [],
+
+        self.codemod.apply(self.execution_context, files_to_check)
+        changes = self.execution_context.get_results(self.codemod.id)
+
+        if input_code == expected:
+            assert not changes
+            return
+
+        assert len(changes) == num_changes
+
+        self.assert_changes(
+            tmpdir,
+            tmp_file_path,
+            input_code,
+            expected,
+            changes[0],
         )
-        codemod_instance = self.initialize_codemod(input_tree)
-        output_tree = codemod_instance.transform_module(input_tree)
 
-        assert output_tree.code == dedent(expected)
+    def assert_changes(  # pylint: disable=too-many-arguments
+        self, root, file_path, input_code, expected, changes
+    ):
+        expected_diff = create_diff(
+            dedent(input_code).splitlines(keepends=True),
+            dedent(expected).splitlines(keepends=True),
+        )
 
-    def assert_dependency(self, dependency: Dependency):
-        assert self.file_context and self.file_context.dependencies == set([dependency])
+        assert expected_diff == changes.diff
+        assert os.path.relpath(file_path, root) == changes.path
+
+        with open(file_path, "r", encoding="utf-8") as tmp_file:
+            output_code = tmp_file.read()
+
+        assert output_code == dedent(expected)
+
+    def run_and_assert_filepath(  # pylint: disable=too-many-arguments
```

Comment:
you could just add this disable to the top of the file instead of for every method

Reply:
Good point, I actually thought about just disabling it globally since it's not my favorite pylint rule. Any thoughts?